Documentation/bootstrapping-guidelines.md

```diff
@@ -0,0 +1,130 @@
+Determine the compatible supported OS to use as host to build. Choose the host with same processor architecture as that of the new targeted platform.
+
+**Stage 1:**
+Build bootstrapping .NET SDK and toolset.
+
+   ``` bash
+   ./prep.sh
+   ```
+
+ Downloads platform-native version of .NET SDK and toolset. If you are building on x64, you will get x64.
+
+   ``` bash
+   ./build.sh
+   ```
+
+Create source-built .NET SDK which will be utilized to bootstrap in stage 2.
+
+Now that you have a stage1 SDK which knows about your new RID, you can build stage2 targeting that RID.
+
+**Stage 2:**
+Extract the source-built .NET SDK and toolset created in stage 1 to build .NET SDK from source. No need to run prep.sh in this stage.
+
+``` bash
+./build.sh --with-sdk /path/to/stage1/sdk --with-packages /path/to/stage1/obj/bin/arch/blob-feed/packages`
+```
+
+**If no compatible OS/platform found**
+
+Use portable SDK (works for .NET 6, but require validation for .NET 7) and follow the steps:
+
+**Stage 0:**
+
+- Get Microsoft portable SDK.
+- Update the RID graph (runtime.json) in the Microsoft-built portable SDK with the same changes you will make below to add your new RID to the RID graph.  This should include a fallback to the portable RID (linux-x64 or similar).
+
+**Stage 1:**
+
+- Before building your stage1 SDK, update the RID graph in source to include your new target RID.  For an example, see <https://github.com/dotnet/runtime/pull/74372>.
+- Build with Stage 0 SDK using `--with-sdk` with your modified portable SDK.
+
+**Stage 2:**
+
+- Now you have a RID-specific SDK that knows about your new RID.
+- Build with Stage 1 SDK as above, using `--with-sdk` and `--with-packages`.
+
+## Building for New Architecture (Using a RID and architecture unknown to .NET)
+
+**RIDs:**
+
+The RID graph or runtime fallback graph is a list of RIDs that are compatible with each other. You can see the list of supported RIDs and the RID graph in the [runtime.json](https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.NETCore.Platforms/src/runtime.json). Learn more about RID catalog in [here](https://learn.microsoft.com/en-us/dotnet/core/rid-catalog#linux-rids).
+
+You will need to update the RID graph to include your new platform and runtime IDs.  See <https://github.com/dotnet/runtime/pull/82382> or <https://github.com/dotnet/runtime/pull/75396> for examples.
+
+Building for unsupported architectures require cross-compilaton on the supported platform. Determine the compatible host to build which provides cross-compilation toolchain.  [IBM has published](https://community.ibm.com/community/user/powerdeveloper/blogs/sapana-khemkar/2023/01/13/cross-build-dotnet7-on-x86-ibm-power?CommunityKey=8cc2a1f0-6307-48cb-9178-ace50920244e) a detailed description of how they successfully built .NET 7 for IBM Power.
```

Comment:
Would it be worth having a section on bootstrap source-build for a new architecture (eg, `linux-s390x`)?

The steps would be roughly like this

- Stage 0
  - Cross compile an SDK (using prebuilts) on x64 for target platform (I don't know the full steps to do this; it's quite long and involved, might include setting up a rootfs for your architecture)
  - Cross compile the runtime repo (on x64 for target platform, generally done as part of previous step) and save the nuget packages, use these to augment the Microsoft-built previously-source-built archive
- Stage 1: Use the cross-compiled SDK, and augmented previously-source-built-archive to build Stage 1 SDK
- Stage 2: Use that to build a State 2 SDK, pointing it to the SDK and previously-source-built archives from Stage 1

Reply:
Thanks Omair, I've added this to the doc.